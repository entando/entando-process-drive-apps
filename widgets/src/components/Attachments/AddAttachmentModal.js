import React from 'react';
import PropTypes from 'prop-types';
import withStyles from '@material-ui/core/styles/withStyles';
import { DropzoneArea } from 'material-ui-dropzone';
import Button from '@material-ui/core/Button';

const styles = {
  buttonWrapper: {
    padding: '10px 0',
    textAlign: 'right',
  },
  dropZone: {
    width: 400,
  },
  dropZoneText: {
    padding: 10,
    fontSize: 15,
  },
};

class AddAttachmentModal extends React.Component {
  state = {
    files: [],
  };

  handleChange = files => {
    this.setState({ files });
  };

  handleUploadButton = () => {
    const { files } = this.state;
    const { onUpload } = this.props;
    onUpload(files);
  };

  render() {
    const { files } = this.state;
    const { classes, onClose } = this.props;

    return (
      <>
        <DropzoneArea
          onChange={this.handleChange}
          dropzoneText="Click or Drag and Drop your attachments"
          dropzoneParagraphClass={classes.dropZoneText}
          dropzoneClass={classes.dropZone}
        />
        <div className={classes.buttonWrapper}>
          <Button variant="outlined" color="default" onClick={onClose}>
            Close
          </Button>{' '}
          <Button
            disabled={!files.length}
            variant="contained"
            color="primary"
            onClick={this.handleUploadButton}
          >
            Upload files
          </Button>
        </div>
      </>
    );
  }
}

AddAttachmentModal.propTypes = {
  classes: PropTypes.shape({
    buttonWrapper: PropTypes.string,
    dropZone: PropTypes.string,
    dropZoneText: PropTypes.string,
  }).isRequired,
  onUpload: PropTypes.func.isRequired,
  onClose: PropTypes.func.isRequired,
};

export default withStyles(styles)(AddAttachmentModal);
