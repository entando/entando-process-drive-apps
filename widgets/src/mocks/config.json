{
  "payload": {
    "code": "pda_task_list",
    "config": {
      "groups": "[{\"label\":\"Administrators\",\"key\":\"Administrators\",\"checked\":true},{\"label\":\"broker\",\"key\":\"broker\",\"checked\":true},{\"label\":\"admin\",\"key\":\"admin\",\"checked\":false},{\"label\":\"rest-all\",\"key\":\"rest-all\",\"checked\":false},{\"label\":\"kiemgmt\",\"key\":\"kiemgmt\",\"checked\":true},{\"label\":\"kie-server\",\"key\":\"kie-server\",\"checked\":true},{\"label\":\"appraiser\",\"key\":\"appraiser\",\"checked\":false},{\"label\":\"manager\",\"key\":\"manager\",\"checked\":true},{\"label\":\"supplier\",\"key\":\"supplier\",\"checked\":true},{\"label\":\"IT\",\"key\":\"IT\",\"checked\":true},{\"label\":\"PM\",\"key\":\"PM\",\"checked\":true}]",
      "options": "[{\"key\":\"newPageOnClick\",\"label\":\"Open new page on table row click\",\"checked\":false},{\"key\":\"showClaim\",\"label\":\"Show Claim Button\",\"checked\":true},{\"key\":\"showComplete\",\"label\":\"Show Complete Button\",\"checked\":true}]",
      "columns": "[{\"name\":\"task-is-skipable\",\"position\":0,\"isVisible\":false},{\"name\":\"task-parent-id\",\"position\":1,\"isVisible\":false},{\"name\":\"task-created-by\",\"position\":2,\"isVisible\":true},{\"name\":\"task-input-data.application.property.saleprice\",\"position\":3,\"isVisible\":true},{\"name\":\"task-input-data.TaskName\",\"position\":4,\"isVisible\":true},{\"name\":\"initiator\",\"position\":5,\"isVisible\":true},{\"name\":\"task-created-on\",\"position\":6,\"isVisible\":true},{\"name\":\"task-priority\",\"position\":7,\"isVisible\":false},{\"name\":\"task-proc-def-id\",\"position\":8,\"isVisible\":true},{\"name\":\"task-form\",\"position\":9,\"isVisible\":true},{\"name\":\"task-skippable\",\"position\":10,\"isVisible\":true},{\"name\":\"task-input-data.application.applicant.name\",\"position\":11,\"isVisible\":true},{\"name\":\"task-actual-owner\",\"position\":12,\"isVisible\":true},{\"name\":\"task-activation-time\",\"position\":13,\"isVisible\":true},{\"name\":\"task-input-data.application.applicant.address\",\"position\":14,\"isVisible\":true},{\"name\":\"task-input-data.application.errors\",\"position\":15,\"isVisible\":true},{\"name\":\"task-name\",\"position\":16,\"isVisible\":true},{\"name\":\"task-input-data.application.property.address\",\"position\":17,\"isVisible\":true},{\"name\":\"task-subject\",\"position\":18,\"isVisible\":true},{\"name\":\"containerId\",\"position\":19,\"isVisible\":true},{\"name\":\"task-container-id\",\"position\":20,\"isVisible\":true},{\"name\":\"processDefinitionId\",\"position\":21,\"isVisible\":true},{\"name\":\"task-input-data.application.applicant.ssn\",\"position\":22,\"isVisible\":true},{\"name\":\"task-description\",\"position\":23,\"isVisible\":true},{\"name\":\"task-input-data.Skippable\",\"position\":24,\"isVisible\":true},{\"name\":\"task-workitem-id\",\"position\":25,\"isVisible\":true},{\"name\":\"task-expiration-time\",\"position\":26,\"isVisible\":true},{\"name\":\"task-input-data.application.property.age\",\"position\":27,\"isVisible\":true},{\"name\":\"task-status\",\"position\":28,\"isVisible\":true},{\"name\":\"task-input-data.application.mortgageamount\",\"position\":29,\"isVisible\":true},{\"name\":\"task-process-instance-id\",\"position\":30,\"isVisible\":true},{\"name\":\"task-input-data.ActorId\",\"position\":31,\"isVisible\":true},{\"name\":\"task-process-id\",\"position\":32,\"isVisible\":true},{\"name\":\"task-input-data.application.property.locale\",\"position\":33,\"isVisible\":true},{\"name\":\"task-input-data.application.amortization\",\"position\":34,\"isVisible\":true},{\"name\":\"task-input-data.application.applicant.creditrating\",\"position\":35,\"isVisible\":true},{\"name\":\"application\",\"position\":36,\"isVisible\":true},{\"name\":\"task-input-data.NodeName\",\"position\":37,\"isVisible\":true},{\"name\":\"task-proc-inst-id\",\"position\":38,\"isVisible\":true},{\"name\":\"task-input-data.application.downpayment\",\"position\":39,\"isVisible\":true},{\"name\":\"task-input-data.application.applicant.annualincome\",\"position\":40,\"isVisible\":true},{\"name\":\"task-id\",\"position\":41,\"isVisible\":true}]",
      "knowledgeSource": "kieStaging",
      "process": "itorders.orderhardware@itorders_1.0.0-SNAPSHOT"
    }
  },
  "metaData": {
    "status": "draft"
  },
  "errors": []
}
