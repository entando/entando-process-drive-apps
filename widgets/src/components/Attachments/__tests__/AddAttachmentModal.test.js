import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import AddAttachmentModal from '../AddAttachmentModal';
import 'mocks/i18nMock';

describe('<AddAttachmentModal />', () => {
  it('renders snapshot correctly', () => {
    const onClose = jest.fn();
    const onUpload = jest.fn();
    const { container } = render(<AddAttachmentModal onClose={onClose} onUpload={onUpload} />);

    expect(container).toMatchSnapshot();
  });

  it('onClose to be called when close button is clicked', () => {
    const onClose = jest.fn();
    const onUpload = jest.fn();
    const { getAllByRole } = render(<AddAttachmentModal onClose={onClose} onUpload={onUpload} />);

    fireEvent.click(getAllByRole('button')[0]);

    expect(onClose).toHaveBeenCalled();
  });
});
