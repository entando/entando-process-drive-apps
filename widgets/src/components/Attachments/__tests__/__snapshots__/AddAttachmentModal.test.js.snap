// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddAttachmentModal /> renders snapshot correctly 1`] = `
<div>
  <div
    aria-disabled="false"
    class="AddAttachmentModal-dropZone-2 DropzoneArea-dropZone-5"
    style="position: relative;"
  >
    <div
      class="DropzoneArea-dropzoneTextStyle-8"
    >
      <p
        class="DropzoneArea-dropzoneParagraph-10 AddAttachmentModal-dropZoneText-3"
      >
        Click or Drag and Drop your attachments
      </p>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root DropzoneArea-uploadIconSize-9"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM14 13v4h-4v-4H7l5-5 5 5h-3z"
        />
      </svg>
    </div>
    <input
      accept="image/*,video/*,application/*"
      autocomplete="off"
      multiple=""
      style="position: absolute; top: 0px; right: 0px; bottom: 0px; left: 0px; opacity: 0.00001; pointer-events: none;"
      type="file"
    />
  </div>
  <div
    class="AddAttachmentModal-buttonWrapper-1"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Close
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
     
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Upload files
      </span>
    </button>
  </div>
</div>
`;
